pytorch_lightning/trainer/supporters.py

```diff
@@ -371,6 +363,56 @@ def __init__(self, loaders: Any, mode: str = 'min_size'):
         if self.mode == 'max_size_cycle':
             self._wrap_loaders_max_size_cycle()
 
+        self.loaders_iter_state_dict = None
+
+    def state_dict(self, num_batches_processed: int):
+        if not fault_tolerant_enabled():
+            return {}
+
+        def state_dict_fn(dataloader: DataLoader, iterator: Iterator) -> Dict:
+            nonlocal num_batches_processed
```

Comment:
```suggestion
```

this should not be necessary unless you want to set a value to it (which you don't here)